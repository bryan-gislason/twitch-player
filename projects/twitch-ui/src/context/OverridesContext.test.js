import PropTypes from "prop-types";
import React, { useContext } from "react";
import { Text } from "react-native";
import { render } from "react-testing-library";
import { OverridesContext, OverridesProvider } from ".";

const SomeComponent = () => {
  const { ChannelPlayer } = useContext(OverridesContext);

  return <ChannelPlayer src="https://www.twitch.tv/" />;
};

const SomeChannelPlayer = ({ src }) => <Text>{src}</Text>;

SomeChannelPlayer.propTypes = {
  src: PropTypes.string.isRequired,
};

describe("OverridesContext", () => {
  let instance;

  describe("without overrides", () => {
    beforeEach(() => {
      instance = render(
        <OverridesProvider>
          <SomeComponent />
        </OverridesProvider>
      );
    });

    it("renders with the default components", () => {
      expect(instance.asFragment()).toMatchSnapshot();
    });
  });

  describe("with overrides", () => {
    beforeEach(() => {
      instance = render(
        <OverridesProvider overrides={{ ChannelPlayer: SomeChannelPlayer }}>
          <SomeComponent />
        </OverridesProvider>
      );
    });

    it("renders with those overrides", () => {
      expect(instance.asFragment()).toMatchSnapshot();
    });
  });
});
