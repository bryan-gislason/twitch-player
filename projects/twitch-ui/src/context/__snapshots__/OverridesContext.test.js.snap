// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverridesContext with overrides renders with those overrides 1`] = `
<DocumentFragment>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    https://www.twitch.tv/
  </div>
</DocumentFragment>
`;

exports[`OverridesContext without overrides renders with the default components 1`] = `<DocumentFragment />`;
